src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/BatchingSqlJournal.cs

```diff
@@ -1205,29 +1203,35 @@ protected virtual void WriteEvent(TCommand command, IPersistentRepresentation pe
             var payloadType = persistent.Payload.GetType();
             var serializer = _serialization.FindSerializerForType(payloadType, Setup.DefaultSerializer);
 
-            string manifest = "";
-            if (serializer is SerializerWithStringManifest)
+            // TODO: hack. Replace when https://github.com/akkadotnet/akka.net/issues/3811
+            Akka.Serialization.Serialization.WithTransport(_serialization.System, () =>
```

Comment:
We don't pass it in because other libraries may not directly have access to when it's set via Akka.Remote - rather it gets parked ambiently in the local thread's storage and then unset when it's done being used. Won't be an issue for multiple actor systems due to the latter - the reference always gets unset and not leaked from the thread. There's more comments on the JVM issue that I ported this from: https://github.com/akka/akka/pull/25068